book/platform/README.md

```diff
@@ -0,0 +1,473 @@
+
+### Writing test cases for a library
+
+The `(libraries)` field in the `hello` dune file is empty since this is a standalone library. Our next step is to define a test case in `tests/dune` for our library.
+
+```scheme
+(test
+ (name hello_test)
+ (libraries alcotest hello))
+```
+
+The `(test)` field builds an executable binary that is run when you invoke `dune runtest`.  In this case, it uses the `tests/hello_test.ml` module to define the test cases and depends on the external `alcotest` library _and_ the 
+locally defined `hello` library.  Once you run the tests, you can find the built artefacts in `_build/default/tests/` in your project checkout.  You can use all the tests you learnt about in [Testing](testing.html){data-type=xref} here, including inline tests.
+
+### Building an executable program
+
+Finally, we want to actually use our hello world from the command-line. This is defined in `bin/dune` in a very similar fashion to test cases.
+
+```scheme
+(executable
+ (name main)
+ (public_name hello)
+ (libraries hello))
+```
+
+Much like libraries, the `(name)` field here has to adhere to OCaml module naming conventions, and the `public_name` field represents the binary name that is installed onto the system and just needs to be a valid Unix or Windows filename.
+
+You can build and execute the command locally using `dune exec` and the public name of the executable:
+
+```sh dir=examples/correct/hello-world
+$ dune exec -- hello
+Hello world!
+```
+
+## Setting up an integrated development environment
+
+Now that we've seen the basic structure of the OCaml project, it's time to setup an integrated development environment. An IDE is particularly useful for use with OCaml due to the extra information you gain from the static type information present in the codebase. A good IDE will provide you with the facilities to browse interface documentation, see inferred types for code, and to jump to the definitions of external modules.
+
+### Using Visual Studio Code
+
+The recommended IDE for newcomers to OCaml is [Visual Studio Code](https://code.visualstudio.com) using the [OCaml Platform plugin](https://marketplace.visualstudio.com/items?itemName=ocamllabs.ocaml-platform).  The plugin uses the Language Server Protocol to communicate with your opam and dune environment. All you need to do is to install the OCaml LSP server via opam:
+
+```
+opam install ocaml-lsp-server
```

Comment:
You'd also want to install `ocamlformat` and `ocamlformat-rpc` to have the full functionalities of the extension.